### PR TITLE
Properly emulate IntrPin field in PCI dev

### PR DESCRIPTION
**Pull request by pfmooney**
This should address #134

### CODE REVIEW

**Review by gjcolombo — APPROVED** [review]

File: propolis/src/hw/pci/device.rs [hunk 1]
```diff
@@ -290,7 +288,19 @@ impl DeviceState {
                 ro.write_u8(self.state.lock().unwrap().reg_intr_line)
             }
             StdCfgReg::IntrPin => {
-                ro.write_u8(self.state.lock().unwrap().reg_intr_pin)
+                if self.lintr_support {
```
Comment (pfmooney) [inline propolis/src/hw/pci/device.rs:291]: Ovmf was grumpy about the hostbus device presenting an interrupt pin.

File: propolis/src/hw/chipset/i440fx.rs [hunk 2]
```diff
@@ -678,6 +678,8 @@ impl Piix3PM {
             ..Default::default()
         })
         .add_custom_cfg(PMCFG_OFFSET as u8, PMCFG_LEN as u8)
+        // ACPI device requires lintr for SCI
+        .add_lintr()
```
Comment (pfmooney) [inline propolis/src/hw/chipset/i440fx.rs:682]: Prior to adding this, `lspci` (on alpine) knew of the interrupt associated with the device thanks to the ACPI table, but was still confused about it due to the missing entry here:
```
Interrupt: pin ? routed to IRQ 9
```

With the entire change, `lspci` is much happier about reporting all of the values (for an attached virtio disk and nic):
```
localhost:~# lspci -vv | grep Interrupt
        Interrupt: pin A routed to IRQ 9
        Interrupt: pin B routed to IRQ 10
        Interrupt: pin B routed to IRQ 11
```

File: propolis/src/hw/pci/device.rs (hunk 1 above)
Comment (pfmooney) [reply propolis/src/hw/pci/device.rs:291]: By "grumpy" I mean blowing an assert and hanging during boot:
```
...
Boot Mode:0
SetPciIntLine: PCI host bridge (00:00.0) should have no interrupts!
ASSERT /home/pmooney/git/uefi-edk2/OvmfPkg/Library/PlatformBootManagerLib/BdsPlatform.c(1128): ((BOOLEAN)(0==1))
```

**Comment by pfmooney** [comment]
Merged in 8259ea3ddf49b23d81e45627f93fb8981713d909

**Pull request closed without merging**